### PR TITLE
Added Command Handler

### PR DESCRIPTION
**Pull request by SagnikPradhan**
(Thanks for sending a pull request! Please make sure you click the link above to view the contribution guidelines, then fill out the blanks below.)

What does this implement/fix? Explain your changes.
---------------------------------------------------
Added Command Handler. Makes it easier to add commands.

Does this close any currently open issues?
------------------------------------------
Not really.

Any relevant logs, error output, etc?
-------------------------------------
Nope.

Any other comments?
-------------------
Nope.

Where has this been tested?
---------------------------
I guess so.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,7 @@
 {
 	"name": "typobot",
 	"dependencies": {
+		"abort-controller": "^3.0.0",
```
Comment (Haxified) [inline package.json]: why?

File: src/utils/commandHandler.ts [hunk 2]
```diff
@@ -0,0 +1,173 @@
+import {
+	Message,
+	Client,
+	BitFieldResolvable,
+	PermissionString,
+	MessageEmbed,
+	EmbedField,
+} from 'discord.js';
+
+interface commandHandlerOptions {
```
Comment (Haxified) [inline src/utils/commandHandler.ts]: should be PascalCase (@Robinlemon)

File: src/utils/commandHandler.ts [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+		commands: { aliases: string[]; description: string }[],
+		message: Message,
+	) => Promise<void>;
+	commandNotFoundFnOrMessage?:
+		| ((message: Message) => Promise<void> | void)
+		| string;
+}
+
+// Command Handler Class
+export class commandHandler {
```
Comment (Haxified) [inline src/utils/commandHandler.ts]: should also be PascalCase

File: src/utils/commandHandler.ts [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+		this.bot.on('message', (message: Message) =>
+			this.handleMessageEvt(message),
+		);
+	}
+
+	/**
+	 * Register a command
+	 * @param command An Instance of Command Class
+	 */
+	registerCommand(command: Command): void {
```
Comment (Haxified) [inline src/utils/commandHandler.ts]: Should be private

File: src/utils/commandHandler.ts [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+						embed.addField("`" + aliases[0] + "`", description),
+					);
+					return await message.channel.send(embed);
+				},
+			}),
+		);
+	}
+}
+
+interface commandOptions {
```
Comment (Haxified) [inline src/utils/commandHandler.ts]: Also PascalCase

File: src/utils/commandHandler.ts [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+	aliases: string[];
+	description: string;
+	privelagesRequired?: BitFieldResolvable<PermissionString>[];
+	command: (message: Message, messageContent: string) => Promise<any>;
+}
+
+// Command Class
+export class Command {
```
Comment (Haxified) [inline src/utils/commandHandler.ts]: Should be abstract

File: package.json (hunk 1 above)
Comment (SagnikPradhan) [reply package.json]: IDK. It said it couldn't find the module

File: src/utils/commandHandler.ts (hunk 6 above)
Comment (SagnikPradhan) [reply src/utils/commandHandler.ts]: Abstract?

File: src/utils/commandHandler.ts (hunk 2 above)
Comment (Haxified) [reply src/utils/commandHandler.ts]: i'm being serious lol

File: package.json (hunk 1 above)
Comment (Haxified) [reply package.json]: Thats weird.

File: src/utils/commandHandler.ts (hunk 4 above)
Comment (SagnikPradhan) [reply src/utils/commandHandler.ts]: Uh. How will it register commands [here](https://github.com/SagnikPradhan/community-bot/blob/master/src/core/Client.ts#L31) ?

Comment (Haxified) [reply src/utils/commandHandler.ts]: ohh nvm

File: src/utils/commandHandler.ts (hunk 6 above)
Comment (Haxified) [reply src/utils/commandHandler.ts]: do you not know what abstract is?

Comment (SagnikPradhan) [reply src/utils/commandHandler.ts]: No

Comment (Haxified) [reply src/utils/commandHandler.ts]: Basically, it means the base class cannot be created. Only classes that extend it can be created.

Comment (Haxified) [reply src/utils/commandHandler.ts]: nvm lmao

**Pull request merged**